### ISSUE TITLE
Repo: Use ESLint v9 for our internal linting 

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#9082]
### Suggestion

Now that #9002 is merged, our `v8` branch supports ESLint v9. Yay! But, #9002 didn't actually bump our _internal_ `devDependency`. https://github.com/typescript-eslint/typescript-eslint/pull/9002#discussion_r1591704965 mentions the three plugins whose lack of ESLint v9 support were blocking us.

But, now that https://eslint.org/blog/2024/05/eslint-compatibility-utilities exists, we should be able to move to ESLint v9 without losing our preferred linting. Which would be great. Accepting PRs to our `v8` branch to bring back the bump of our `"eslint"` `devDependency` from 8 to 9.

💖

**Comment by fredguest** [comment]
Along the same lines, if a project depends on the plugin and the parser as in:
```
"@typescript-eslint/eslint-plugin": "^8.0.0-alpha.10",
"@typescript-eslint/parser": "^8.0.0-alpha.10",
```
you get the following error:
```
Could not resolve dependency:
peer @typescript-eslint/parser@"^7.0.0" from @typescript-eslint/eslint-plugin@8.0.0-alpha.10
```
which is being triggered by the peer dependency on line 105 of the package.json
https://www.npmjs.com/package/@typescript-eslint/eslint-plugin/v/8.0.0-alpha.10?activeTab=code

**Comment by bradzacher** [comment]
@fredguest https://github.com/typescript-eslint/typescript-eslint/issues/9087

**Comment by fredguest** [comment]
@bradzacher cheers 👍
